### DISCUSSION TITLE
Top Lang custom_title parameter is broken

### DISCUSSION THREAD

**Discussion by galexbh** [anuraghazra/github-readme-stats#1949] [Q&A] [answered]
Setting the custom_title parameter causes an error in top lang

It used to work perfectly as shown in the following code

`<a href="https://github.com/anuraghazra/github-readme-stats">
<img align="center" src="https://github-readme-stats.vercel.app/api/top-langs/?username=galexbh&bg_color=0D1117&text_color=C9D1D9&hide_border=true&custom_title=💖 Most Used Languages 🐈&hide=VBA&hide_title=false&exclude_repo=annotations,Machine-learning-course&langs_count=11&layout=compact" />
</a>`

<a href="https://github.com/anuraghazra/github-readme-stats">
<img align="center" src="https://github-readme-stats.vercel.app/api/top-langs/?username=galexbh&bg_color=0D1117&text_color=C9D1D9&hide_border=true&custom_title=💖 Most Used Languages 🐈&hide=VBA&hide_title=false&exclude_repo=annotations,Machine-learning-course&langs_count=11&layout=compact" />
</a>

<br>

Here it works perfectly if I remove the custom_title

`<a href="https://github.com/anuraghazra/github-readme-stats">
<img align="center" src="https://github-readme-stats.vercel.app/api/top-langs/?username=galexbh&bg_color=0D1117&text_color=C9D1D9&hide_border=true&hide=VBA&hide_title=false&exclude_repo=annotations,Machine-learning-course&langs_count=11&layout=compact" />
</a>`

<a href="https://github.com/anuraghazra/github-readme-stats">
<img align="center" src="https://github-readme-stats.vercel.app/api/top-langs/?username=galexbh&bg_color=0D1117&text_color=C9D1D9&hide_border=true&hide=VBA&hide_title=false&exclude_repo=annotations,Machine-learning-course&langs_count=11&layout=compact" />
</a>

**Comment by rickstaa** [comment, marked as answer]
@galexbh Thanks for your bug report. I checked the cards, and I can not reproduce the following syntax:

[![Top Langs](https://github-readme-stats.vercel.app/api/top-langs/?username=anuraghazra&custom_title=💖Most%20Used%20Languages🐈)](https://github.com/anuraghazra/github-readme-stats)

**With custom_title**

```md
[![Top Langs](https://github-readme-stats.vercel.app/api/top-langs/?username=anuraghazra&custom_title=💖Most%20Used%20Language🐈)](https://github.com/anuraghazra/github-readme-stats)
````

**Without custom_title**

[![Top Langs](https://github-readme-stats.vercel.app/api/top-langs/?username=anuraghazra&custom_title=test)](https://github.com/anuraghazra/github-readme-stats)

```md
[![Top Langs](https://github-readme-stats.vercel.app/api/top-langs/?username=anuraghazra&custom_title=test)](https://github.com/anuraghazra/github-readme-stats)
````

I think your problem is you use spaces in a URL which is not allowed. You should use percent-encoding for special characters (see https://en.wikipedia.org/wiki/Percent-encoding). You can use [this stack question](https://stackoverflow.com/questions/7109143/what-characters-are-valid-in-a-url to check which characters can be used without percent-encoding.

**Reply by galexbh** [reply]
Thank you very much, with the %20 it has been solved.

I find it strange that I had it for months without the percentage notation and it worked properly. But it's ok, it's fixed now.

Thank you very much 💜

**Reply by rickstaa** [reply]
No problem. I have no idea how it could break since we did not touch the relevant code in the last months. But glad your problem is solved. 👍 Could you please accept my answer such that this question is listed as solved 🤓.
